automation_tools/__init__.py

```diff
@@ -1655,16 +1656,176 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
-def host_pings(host, timeout=15):
+# =============================================================================
+# Satellite and Capsule Upgrade
+# =============================================================================
+def copy_ssh_key(from_host, to_host):
+    """This will generate(if not already) ssh-key on from_host
+    and copy that ssh-key to to_host.
+
+    Beware that both hosts should have authorized key added
+    for test-running host.
+
+    :param from_host: A string. Hostname on which the key to be generated and
+        to be copied from.
+    :param to_host: A string. Hostname on to which the ssh-key will be copied.
+
+    """
+    execute(lambda: run('[ ! -f ~/.ssh/id_rsa.pub ] && ssh-keygen '
+                        '-f ~/.ssh/id_rsa -t rsa -N \'\''), host=from_host)
+    if execute(lambda: run('[ -f ~/.ssh/id_rsa.pub ]; '
+                           'echo $?'), host=from_host)[from_host] == 0:
+        tmp_path = tempfile.mkstemp()[1]
+        local('scp root@{0}:~/.ssh/id_rsa.pub {1}'.format(from_host, tmp_path))
+        execute(lambda: run('[ ! -f ~/.ssh/authorized_keys ] && '
+                            'touch ~/.ssh/authorized_keys'), host=to_host)
+        local("cat {0} | ssh -o 'StrictHostKeyChecking no' root@{1} "
+              "'cat >> .ssh/authorized_keys'".format(tmp_path, to_host))
+        os.remove(tmp_path)
+
+
+def capsule_certs_generate(capsule=None, path=None):
```

Comment:
Yeah Its a almost similar but the only advantage of this function is, It copies the generated cert into capsule.